commit 9052df55b0c73c2e91441cbcf17509ec7b054859
Author: Isaac Davis <isaac.davis@joyent.com>
Date:   2019-09-09T11:25:48+00:00 (4 weeks ago)
    
    MANTA-4535 `manta-adm update` breaks when deploying Prometheus if CNS allow_transfer metadata does not exist yet

diff --git a/lib/deploy.js b/lib/deploy.js
index 10c6508..8f76870 100644
--- a/lib/deploy.js
+++ b/lib/deploy.js
@@ -742,7 +742,9 @@ Deployer.prototype.addCnsAllowTransfer = function addCnsAllowTransfer(cb)
 	 * Update the cns service with the vm's admin IP, if necessary.
 	 */
 	function updateCnsSvc(ctx, next) {
-		var existingIps = ctx.cnsSvc.metadata.allow_transfer;
+		var allow_transfer = ctx.cnsSvc.metadata.allow_transfer;
+		var existingIps =
+		    allow_transfer === undefined ? [] : allow_transfer;
 		if (existingIps.indexOf(ctx.ip) > -1) {
 			log.info({
 				svcname: self.svcname,
